tests/units/convertToType.ts

```diff
@@ -0,0 +1,35 @@
+import { convertToType } from "../../src/helpers/types";
+
+describe("convertToType", () => {
+  let globalCounter = 0;
+
+  class SampleTarget {
+    constructor(public test: number) {
+      globalCounter++;
+    }
+  }
+
+  beforeEach(() => {
+    globalCounter = 0;
+  });
+
+  it("should properly convert a object to the defined target", async () => {
+    const testObject = {
+      test: 4,
+    };
+
+    const convertedObject = convertToType(SampleTarget, testObject);
+    expect(convertedObject).toBeDefined();
+    expect(convertedObject).toHaveProperty("test");
+    expect(globalCounter).toBe(1);
```

Comment:
Missing check for `instanceOf` - for now it can just return the `data` and the test passes.
We have to check if it really converts the object to a class instance.